sqlglot/parser.py

```diff
@@ -5911,3 +5919,50 @@ def _replace_lambda(
                     else:
                         column.replace(dot_or_id)
         return node
+
+    def _parse_truncate(self) -> t.Optional[exp.Truncate] | exp.Expression:
+        # Not to be confused with TRUNCATE(number, decimals) function call
+        if self._match(TokenType.L_PAREN):
+            return self._parse_function_call(anonymous=True)
+
+        # Clickhouse supports TRUNCATE DATABASE as well
+        is_database = self._match(TokenType.DATABASE)
+
+        self._match(TokenType.TABLE)
+
+        exists = self._parse_exists(not_=False)
+
+        def _parse_truncate_tables() -> t.Optional[exp.Expression]:
+            this = self._parse_table(schema=True, is_db_reference=is_database)
+
+            # Postgres has wildcard (table) suffix operator
+            self._match_text_seq("*")
+
+            return this
+
+        expressions = self._parse_csv(_parse_truncate_tables)
+
+        cluster = self._parse_id_var() if self._match_text_seq("ON", "CLUSTER") else None
+
+        if self._match_text_seq("RESTART", "IDENTITY"):
+            identity = "RESTART"
+        elif self._match_text_seq("CONTINUE", "IDENTITY"):
+            identity = "CONTINUE"
+        else:
+            identity = None
+
+        if self._match_text_seq("CASCADE") or self._match_text_seq("RESTRICT"):
+            option = self._prev.text
+        else:
+            option = None
+
+        return self.expression(
```

Comment:
Is it guaranteed that this method parses the complete / all specs? If not, we should ensure that if there's a `self._curr` here we'll use the `Command` fallback parser.

Reply:
Not all dialects were covered in this PR so added a `exp.Command` fallback case